pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -125,11 +125,13 @@ func (r *reconcilerBase) upsertServiceAccount(
 		if len(refs) > 0 {
 			childSA.OwnerReferences = refs
 		}
-		// Update annotation when Workload Identity is enabled on a GKE cluster.
+		// Set annotation when impersonating a GSA on a Workload Identity enabled cluster.
 		// In case, Workload Identity is not enabled on a cluster and spec.git.auth: gcpserviceaccount,
 		// the added annotation will be a no-op.
```

Comment:
Would you update this comment as well given that the added annotation won't be no-op in some cases and needs to be removed?

Reply:
done